cmd/csi-rclone-plugin/main.go

```diff
@@ -84,59 +86,70 @@ func main() {
 	os.Exit(0)
 }
 
-func handleNode() {
-	d := rclone.NewDriver(nodeID, endpoint)
-	ns, err := rclone.NewNodeServer(d.CSIDriver, cacheDir, cacheSize)
-	if err != nil {
-		panic(err)
-	}
-	meters = append(meters, ns.Metrics()...)
-	d.WithNodeServer(ns)
-	err = d.Run()
-	if err != nil {
-		panic(err)
-	}
-}
-
-func nodeCommandLineParameters(runCmd *cobra.Command) {
+func NodeCommandLineParameters(runCmd *cobra.Command, meters *[]metrics.Observable, nodeID, endpoint, cacheDir, cacheSize *string) error {
 	runNode := &cobra.Command{
 		Use:   "node",
 		Short: "Start the CSI driver node service - expected to run in a daemonset on every node.",
-		Run: func(cmd *cobra.Command, args []string) {
-			handleNode()
+		RunE: func(cmd *cobra.Command, args []string) error {
+			//Pointers are passed by value, so we use a pointer to a pointer to be able to retrieve the allocated server in the
+			//run closure
+			var nsDoublePointer **rclone.NodeServer
```

Comment:
question: Are you sure the double pointer here is really necessary? If the pointer to `rclone.NodeServer` is passed by value I think that means that the pointer still points to the same thing but that address is just copied in memory. So we get a new pointer but it points to the same thing.